store/dataStore.ts

```diff
@@ -1,17 +1,33 @@
 import { create } from 'zustand';
+import { produce } from 'immer';
+import { IUser } from '@/api/IapiFunctions';
 
-interface Example {
-    id: number;
-    name: string;
+interface IDataStoreState {
+  user: IUser;
 }
-interface ExampleStore {
-    examples: Example[];
+
+interface IDataStoreAction {
+  updateUser: (id: IUser['id'], email: IUser['email']) => void;
+  reset: () => void;
 }
 
-const dataStore = () => ({
-    examples: [],
-})
+export interface DataStore extends IDataStoreState, IDataStoreAction {}
 
-const useDataStore = create<ExampleStore>(dataStore)
+const initialState: IDataStoreState = {
+  user: {
+    id: '',
+    email: '',
+  },
+};
 
-export default useDataStore;
\ No newline at end of file
+export const useDataStore = create<DataStore>((set) => ({
+  ...initialState,
+  updateUser: (id, email) =>
+    set(
+      produce((state: IDataStoreState) => {
+        state.user.id = id;
+        state.user.email = email;
+      }),
+    ),
+  reset: () => set(initialState),
```

Comment:
Change this to `resetUser`

Reply:
✅ Done!

`resetUser: () => set({ user: initialState.user })`